tensorflow/compiler/xla/literal_test.cc

```diff
@@ -1228,6 +1229,12 @@ TEST_F(LiteralUtilTest, F16) {
   half h1(1.0f);
   half h2(2.0f);
   auto m2 = LiteralUtil::CreateR2<half>({{h1, h2}, {h2, h1}});
+  if (!tensorflow::port::kLittleEndian) {
+    auto array_16 = reinterpret_cast<uint16_t*>(m2.data<half>().data());
+    for (int i = 0; i < 4; i++) {
+        array_16[i] = BYTE_SWAP_16(array_16[i]);
+    }
+  }
   const char* d2 = reinterpret_cast<const char*>(m2.data<half>().data());
   EXPECT_EQ(d2[0], 0);
   EXPECT_EQ(d2[1], 0x3C);
```

Comment:
It might be nicer to instead do:
```
  const uint16_t* d2 = reinterpret_cast<const uint16_t*>(m2.data<half>().data());
  EXPECT_EQ(d2[0], 0x3C00);
  EXPECT_EQ(d2[1], 0x4000);
  EXPECT_EQ(d2[2], 0x4000);
  EXPECT_EQ(d2[3], 0x3C00);
```

Reply:
@majnemer Yes, I agree. But since some similar codes in `leteral_test.cc` also use `const char* d`, I assume we can keep it unchanged? Thanks.

Reply:
Hi @majnemer , yes, you're right. This is a nicer way to fix this endianness issue. I've updated the code changes. Please take a look when you're available. Thanks!

Reply:
LGTM, thanks!